src/catalog/distribute_iterator.cc

```diff
@@ -227,22 +227,30 @@ void DistributeWindowIterator::SeekToFirst() {
         return;
     }
     for (const auto& kv : *tables_) {
-        it_.reset(kv.second->NewWindowIterator(index_));
-        it_->SeekToFirst();
-        if (it_->Valid()) {
-            cur_pid_ = kv.first;
-            return;
+        auto it = kv.second->NewWindowIterator(index_);
+        if (it != nullptr) {
+            it->SeekToFirst();
+            if (it->Valid()) {
+                DLOG(INFO) << "first pos in local: pid=" << kv.first;
+                it_.reset(it);
+                cur_pid_ = kv.first;
+                return;
+            }
+            delete it;
         }
     }
     for (const auto& kv : tablet_clients_) {
         uint32_t count = 0;
-        cur_pid_ = kv.first;
-        kv_it_ = kv.second->Traverse(tid_, cur_pid_, index_name_, "", 0, FLAGS_traverse_cnt_limit, count);
-        if (kv_it_ && kv_it_->Valid()) {
-            response_vec_.emplace_back(kv_it_->GetResponse());
+        auto it = kv.second->Traverse(tid_, kv.first, index_name_, "", 0, FLAGS_traverse_cnt_limit, count);
```

Comment:
the `Seek` has the same problem. it's better to check it_ validation in `GetKey` and `GetValue` function

Reply:
we should ensure only zero or one of `it_` and `kv_it_` is not null.
So the check here is whether `it_` and `kv_it_` both not null.
It is `it_` and `kv_it_` 's duty to check valid itself before `GetKey` or `GetValue`

Reply:
we should improve type safety for iterators, in both definition and implementation. E.g the `GetKey` method of `WindowIteraotr` should mark as `const`